cmd/archive/unpack.go

```diff
@@ -36,26 +38,47 @@ const (
 	FailedChroot
 	FailedChdirAfterChroot
 	FailedInvokeUnpack
+	FailedApplyLayer
 )
 
-func main() {
-	ctx := context.Background()
-	op := trace.NewOperation(ctx, "Unpack") // TODO op ID?
-	op.Debugf("New unpack operation created")
-
-	if len(os.Args) != 4 {
-		op.Errorf("Wrong number of arguments passed to unpack binary")
-		os.Exit(WrongArgumentCount)
-	}
-
-	root := os.Args[2]
+// Calls our custom unpack method inside of the chroot
+func customUnpack(op trace.Operation) {
```

Comment:
Not sure it makes sense to do that in this PR since it is not your change, but referring os.Args[3] inside this function is just not even close to ok. I am only making this comment to point that out, not to change it.